internal/server/handlers/validation/database_cluster.go

```diff
@@ -575,15 +576,21 @@ func validateDBEngineVersionUpgrade(newVersion, oldVersion string) error {
 	if semver.Compare(newVersion, oldVersion) < 0 {
 		return errDBEngineDowngrade
 	}
-	// We will not allow major upgrades.
-	// Major upgrades are handled differently for different operators, so for now we simply won't allow it.
-	// For example:
-	// - PXC operator allows major upgrades.
-	// - PSMDB operator allows major upgrades, but we need to handle FCV.
-	// - PG operator does not allow major upgrades.
-	if semver.Major(oldVersion) != semver.Major(newVersion) {
+	// We will not allow major upgrades for PXC and PG.
+	// - PXC: Major upgrades are not supported.
+	// - PG: Major upgrades are in technical preview. https://docs.percona.com/percona-operator-for-postgresql/2.0/update.html#major-version-upgrade
+	if engineType != everestv1alpha1.DatabaseEnginePSMDB && semver.Major(oldVersion) != semver.Major(newVersion) {
 		return errDBEngineMajorVersionUpgrade
 	}
+
+	// It's fine to ignore the errors here because we have already validated the version.
+	newMajorInt, _ := strconv.Atoi(semver.Major(newVersion)[1:])
+	oldMajorInt, _ := strconv.Atoi(semver.Major(oldVersion)[1:])
+	// We will not allow major upgrades if the versions are not sequential.
+	if newMajorInt-oldMajorInt > 1 {
```

Comment:
what if newMajor version < oldMajor version?

Reply:
looks like this case is validated earlier in the function, the `errDBEngineDowngrade` will be returned